src/main/java/org/graylog2/gelfclient/GelfConfiguration.java

```diff
@@ -33,12 +34,23 @@
     private int sendBufferSize = -1;
 
     /**
+     * Creates a new configuration with the given hostname and port.
+     *
+     * @param hostname The hostname of the GELF-enabled server
+     * @param port The port of the GELF-enabled server
+     */
+    public GelfConfiguration(final String hostname, final int port) {
```

Comment:
This constructor should throw an IllegalArgumentException if either the hostname or the port number is invalid.

Proposed checks:
* `hostname` not null
* `hostname` not blank (empty after trimming)
* `port` between 1 and 65535 (while 0 is a valid port number, it doesn't make sense here).